language-support/hs/bindings/src/DA/Ledger/Services/CommandService.hs

```diff
@@ -25,13 +25,10 @@ submitAndWait commands =
         let LL.CommandService{commandServiceSubmitAndWaitForTransactionId=rpc} = service
         let request = LL.SubmitAndWaitRequest (Just (lowerCommands commands)) noTrace
         rpc (ClientNormalRequest request timeout emptyMdm)
+            >>= unwrapWithInvalidArgument
             >>= \case
-            ClientNormalResponse LL.SubmitAndWaitForTransactionIdResponse{} _m1 _m2 _status _details -> do
-                return $ Right ()
-            ClientErrorResponse (ClientIOError (GRPCIOBadStatusCode StatusInvalidArgument details)) ->
-                return $ Left $ show $ unStatusDetails details
-            ClientErrorResponse e ->
-                fail (show e)
+            Right LL.SubmitAndWaitForTransactionIdResponse{} -> return $ Right ()
```

Comment:
You can simply this case to  `bimap (show . unStatusDetails) (const ())` which makes it easier to see what is going on here.

Reply:
Didn't need `bimap`, just `fmap`, after the change for your following comment